dashboard/app/models/async_progress_handler.rb

```diff
@@ -0,0 +1,30 @@
+# An SQS messages handler for asynchronous user level progress operations. These
+# include activities and user_level updates.
+
+class AsyncProgressHandler
+
+  # Returns a thread-local SQS queue for handle asynchronous user progress operations
+  # (Activity and UserLevel updates.) The queue is hread-local because the SQS client
+  # is not thread-safe.
+  def AsyncProgressHandler.progress_queue
+    Thread.current['progress_queue'] ||=
+        SQS::SQSQueue.new(Aws::SQS::Client.new, CDO.activity_queue_url)
+  end
+
+  def handle(messages)
+    ActiveRecord::Base.transaction do
+      messages.each do |message|
+        op = JSON.parse(message.body)
```

Comment:
if we're going to want to do this more than twice and don't want this to turn into a giant case statement, could just be something like:
````
model = op['model'].constantize 
if model.respond_to? :handle_async_op
  model.send(:handle_async_op, op)
else
  raise "Unexpected model #{op['model']} in #{op}"
end
````
I think this will raise NameError: uninitialized constant if op['model'] doesn't exist -- you could catch this and raise the same RuntimeError: Unexpected model if you want, but I assume that since the only reason this exception will happen is if our code is wrong you mostly want it to raise something so you can get a stack trace...